api/app/weather_models/precip_rdps_model.py

```diff
@@ -0,0 +1,79 @@
+
+RDPS_S3_PREFIX = f"weather_models/{ModelEnum.RDPS.lower()}/"
+
+
+async def generate_24_hour_accumulating_precip_raster(current_time: datetime):
+    """
+    Given a UTC datetime, grab the raster for that date
+    and the date for 24 hours before to compute the difference.
+    """
+    (yesterday_key, today_key) = get_raster_keys_to_diff(current_time)
+    day_data = await read_into_memory(f"{RDPS_S3_PREFIX}/{today_key}")
+
+    yesterday_time = current_time - timedelta(days=1)
+    yesterday_data = await read_into_memory(f"{RDPS_S3_PREFIX}/{yesterday_key}")
+    if day_data is None or yesterday_data is None:
+        raise ValueError("No precip raster data for %s or %s", today_key, yesterday_key)
+
+    later_precip = TemporalPrecip(timestamp=current_time, precip_amount=day_data)
+    earlier_precip = TemporalPrecip(timestamp=yesterday_time, precip_amount=yesterday_data)
+    return compute_precip_difference(later_precip, earlier_precip)
+
+
+def get_raster_keys_to_diff(timestamp: datetime):
+    """
+    To maintain the invariant that only the same model runs are diffed, we need to grab the latest
+    model run that has data greater than 24 hours ago, but less than 36 hours ago.
+    """
+    target_model_run_date = timestamp - timedelta(hours=24)
+    key_prefix = f"weather_models/{ModelEnum.RDPS.lower()}/{target_model_run_date.date().isoformat()}"
+    # From earlier model run, get the keys for 24 hours before timestamp and the timestamp to perform the diff
+    earlier_key = f"{key_prefix}/"
+    later_key = f"{key_prefix}/"
+    if target_model_run_date.hour != 0 and target_model_run_date.hour != 12:
+        # we're not looking at a run hour, so prefix key with computed path
+        earlier_key = (
+            earlier_key
+            + f"computed/{target_model_run_date.hour:02d}/precip/{compose_computed_rdps_filename(target_model_run_date, target_model_run_date.hour, target_model_run_date.hour)}"
+        )
+        later_key = later_key + f"computed/{timestamp.hour:02d}/precip/{compose_computed_rdps_filename(timestamp, timestamp.hour, timestamp.hour)}"
+    else:
+        earlier_key = earlier_key + f"{target_model_run_date.hour:02d}/precip/{compose_rdps_filename(target_model_run_date, target_model_run_date.hour, target_model_run_date.hour)}"
+        later_key = later_key + f"{timestamp.hour:02d}/precip/{compose_rdps_filename(timestamp, timestamp.hour, timestamp.hour)}"
+    return (earlier_key, later_key)
+
+
+def compute_precip_difference(later_precip: TemporalPrecip, earlier_precip: TemporalPrecip):
+    """
+    Simple function to compute difference between later and earlier precip values
+    to be vectorized with numba.
+    """
+    if not later_precip.is_after(earlier_precip):
+        raise ValueError("Later precip value must be after earlier precip value")
+    return vectorized_diff(later_precip.precip_amount, earlier_precip.precip_amount)
+
+
+def _diff(value_a: float, value_b: float):
+    return value_a - value_b
```

Comment:
Should we add logic here to make sure we don't end up with negative numbers due to floating point errors as E mentioned during the sprint review? Could also be a separate task.

Reply:
Pulled out a separate task here since we'll need to figure out how to do this with numba's vectorize: https://github.com/bcgov/wps/issues/3753